keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/DoubleVertex.java

```diff
@@ -23,7 +23,7 @@
      */
     protected abstract DualNumber calculateDualNumber(Map<Vertex, DualNumber> dualNumbers);
 
-    public DualNumber getDualNumber() {
+    public final DualNumber getDualNumber() {
```

Comment:
Nice. Is this to stop people implementing it by mistake instead of `calculateDualNumber`?